js/plugin/RoutingPathQuality.js

```diff
@@ -60,6 +60,159 @@ BR.RoutingPathQuality = L.Control.extend({
+                                        break;
+                                    case 'grade3':
+                                        if (surface === null) {
+                                            surface = 0.4;
+                                        } else {
+                                            surface *= 0.8;
+                                        }
+                                        break;
+                                    case 'grade4':
+                                        if (surface === null) {
+                                            surface = 0.1;
+                                        } else {
+                                            surface *= 0.6;
+                                        }
+                                        break;
+                                    case 'grade5':
+                                        if (surface === null) {
+                                            surface = 0.01;
+                                        } else {
+                                            surface *= 0.4;
+                                        }
+                                        break;
+                                }
+                            }
+
+                            if (surface !== null) {
+                                // modifier for surface quality
+                                switch (data.get('smoothness')) {
+                                    case 'excellent':
+                                        surface = Math.max(surface * 1.1, 1.0);
+                                        break;
+                                    case 'good':
+                                        surface = Math.max(surface * 1.05, 1.0);
+                                        break;
+                                    case 'intermediate':
+                                        surface *= 0.9;
+                                        break;
+                                    case 'bad':
+                                        surface *= 0.7;
+                                        break;
+                                    case 'very_bad':
+                                        surface *= 0.5;
+                                        break;
+                                    case 'horrible':
+                                        surface *= 0.4;
+                                        break;
+                                    case 'very_horrible':
+                                        surface *= 0.2;
+                                        break;
+                                    case 'impassable':
+                                        surface *= 0.01;
+                                        break;
+                                }
+                            }
+
+                            // limit normal values 0-0.9 so 1.0 can be unknown
+                            const final = surface === null ? 1.0 : surface * 0.9;
```

Comment:
`surface=asphalt` + `smoothness=excellent` looks more like unknown, probably because `final` = ~0.99 (1 * 1.1 * 0.9), which is closer to gray for unknown than green, example:
- http://localhost:3000/#map=16/47.6917/9.0687/standard,route-quality&lonlats=9.062669,47.692158;9.070222,47.690237
- [way ‪147345349‬](https://www.openstreetmap.org/way/147345349)